### ISSUE TITLE
Extraction probleam

### ISSUE THREAD

**Issue by osmnbtrclk** [OmniBlade/ccmix#1]
When I try --extract --file all program says "generates an ID conflict with existing entry" but if I add --game ra2 says this time "Extraction failed" how can I fix thath.

**Comment by osmnbtrclk** [comment]
![image](https://cloud.githubusercontent.com/assets/11875855/22902571/32424fa0-f23e-11e6-9571-c9e5fc6111c4.png)
When I try this way, program chrashing.

**Comment by ghost** [comment]
I am getting the exact same problem too.

**Comment by OmniBlade** [comment]
Is this with a build from the crypto branch or master? The crypto branch is the most up to date, if that has issues I will look into them.

**Comment by osmnbtrclk** [comment]
Im downloaded from ppm, I cant compile master and crypto with vs2015.

**Comment by osmnbtrclk** [comment]
Im tried file>>new>>project from existing code and selected scr folder after thath pressed build button but giving this errors in vs2015 
![image](https://cloud.githubusercontent.com/assets/11875855/22981455/22be1506-f3a5-11e6-810e-a16568e02102.png)

**Comment by osmnbtrclk** [comment]
When I try use MinGW giving this error.
![image](https://cloud.githubusercontent.com/assets/11875855/22983264/4465a516-f3aa-11e6-9100-757df12456b5.png)

**Comment by OmniBlade** [comment]
The crypto branch has a cmake configuration, run it through cmake and it will generate a solution for msvc.

**Comment by osmnbtrclk** [comment]
When I try cmake gui giving this error.
![image](https://cloud.githubusercontent.com/assets/11875855/23030692/2dc2aa16-f477-11e6-8a07-7b3dd0d341b8.png)

**Comment by OmniBlade** [comment]
Yeah, because you need to build crypto++ first and then point cmake at it with those variables.

**Comment by osmnbtrclk** [comment]
Im compiled crypto++ and selected ,n CRYPTOPP_ROOT_DIR section but this time giving libcrypto++ not found .
![image](https://cloud.githubusercontent.com/assets/11875855/23065370/df59f3fa-f51d-11e6-9415-2d62046d050d.png)

**Comment by osmnbtrclk** [comment]
Any chance to send compiled version to me sir ? If is possible.

**Comment by osmnbtrclk** [comment]
Im fixed probleam, cmake build this ccmix.sln but I cant compile with v2015 Im doing something wrong.
![image](https://cloud.githubusercontent.com/assets/11875855/23091774/7290d4b0-f5c6-11e6-855f-c8cf6a628f7c.png)

**Comment by osmnbtrclk** [comment]
When I try compile with make command giving this error.
![image](https://cloud.githubusercontent.com/assets/11875855/23094163/632fde54-f5fc-11e6-814f-0cb20da30aec.png)

**Comment by osmnbtrclk** [comment]
I fixed previous probleam but I have a another one.
![image](https://cloud.githubusercontent.com/assets/11875855/23096711/cb00d640-f62a-11e6-8102-ee7a29109c3a.png)

**Comment by osmnbtrclk** [comment]
my probleam still continues only export %50 of files other %50 still giving "extraction faild" with crytopp branch.

![image](https://cloud.githubusercontent.com/assets/11875855/23106414/db590884-f6f5-11e6-8b64-0261986495f5.png)

**Comment by osmnbtrclk** [comment]
When I try --extract --file (fileid) program says "generates an ID conflict with existing entry" but if I add --game ra2 says this time "Extraction failed" with crypto branch. When I use --list command list 7818 file but only export nealry 3000 file.

**Comment by OmniBlade** [comment]
Hmm, its been a while since i built and tested this, I'm busy with other projects at the moment, but I'll look into it when I get chance. If you figure it out in the mean time, feel free to submit a pull request.

**Comment by osmnbtrclk** [comment]
Okay thanks again. I'll hipe you can find a time for looking code again. If I find usefull anything I'll send pull request.

**Comment by SamKeightley** [comment]
I was having the same problem too but though it has been so long that this tool was dead. Glad to see that it's not.

**Comment by osmnbtrclk** [comment]
I cant figure out but I suppose this probleam happening because of blowfish.This can be right ? If mix file protected with blowfish maybe program cannot reverse id to file but still I can extract files of %50.

**Comment by OmniBlade** [comment]
Are the extracted files correct? If so then it can't be an issue with blowfish because if it was you would get nothing or garbage extracted. Blowfish protects the header and if it can decrypt part of the header it can decrypt it all as the key doesn't change.

**Comment by osmnbtrclk** [comment]
Yes files are correct, when I try only use --extract command extract files 1000+/1385 but still few files missing.

**Comment by andersevenrud** [comment]
Any progress on this? I'm having the exact same issues. I also had to add `-Wno-variadic-macros` for this to compile, or else I got this:

```
/home/anders/Projects/ccmix/src/mixid.cpp:43:107: error: narrowing conversion of ‘3009837614’ from ‘unsigned int’ to ‘int32_t {aka int}’ inside { } [-Wnarrowing]
```

**Comment by OmniBlade** [comment]
Variadic macros have absolutely nothing to do with that error, it just doesn't like the CRC table being signed and filled with hex values that the compiler is treating as unsigned.

I've not made any attempt at fixing the issue BTW, this project can be considered unmaintained for now, as I don't have time to look into it.
